lib/http-invocation.js

```diff
@@ -148,7 +148,13 @@ HttpInvocation.prototype._processArg = function(req, verb, query, accept) {
     // default to query string for GET
     if (val !== undefined) {
       query = query || {};
-      query[name] = val;
+
+      if ((accept.type === 'object' || accept.type === 'string') &&
+        typeof val === 'object') {
+        query[name] = JSON.stringify(val);
+      } else {
+        query[name] = val;
+      }
```

Comment:
@horiaradu : This needs to be applied for `query` case in here as well:
https://github.com/strongloop/strong-remoting/blob/master/lib/http-invocation.js#L127-L132

Better solution would be to extract it in a common function.